### PR TITLE
verify ticket tailor hmac signature

### PR DESCRIPTION
**Pull request by michaelwmcnamara**
## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: handlers/ticket-tailor-webhook/test/verifySignature.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { hasMatchingSignature } from '../src/verifySignature';
+import { HmacKey } from '../src/hMacKey';
+
+/**
+ * This is a unit test, it can be run by the `pnpm test` command, and will be run by the CI/CD pipeline
+ * WARNING: Do not alter key or mock records in this file as it will stop the test from working.
+ */
+
+const mockKey: HmacKey = {
+	secret: '9dn189d53me1ania7d73a45d5de4674d',
```
[the remark this answers is not in the document]
Comment (michaelwmcnamara) [reply handlers/ticket-tailor-webhook/test/verifySignature.test.ts:11]: This is a mocked key and is not one in use anywhere in the organisation.

File: handlers/ticket-tailor-webhook/src/verifySignature.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export const hasMatchingSignature = (
+	record: SQSRecord,
+	validationSecret: HmacKey,
+): boolean => {
+	const signatureWithTs =
+		record.messageAttributes['tickettailor-webhook-signature']?.stringValue;
+
+	if (typeof signatureWithTs === 'string') {
```
Comment (MaelGNM) [inline handlers/ticket-tailor-webhook/src/verifySignature.ts:20]: Minor comment for readability and understanding, could we rename all `ts` to `timestamp` so it's clearer what we're doing?

File: handlers/ticket-tailor-webhook/src/verifySignature.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		const signature = signatureWithTs.split(',')[1]?.split('=')[1];
+		const ts = signatureWithTs.split(',')[0]?.split('=')[1];
```
Comment (MaelGNM) [inline handlers/ticket-tailor-webhook/src/verifySignature.ts:22]: This looks a bit like magic. Maybe adding a comment above to show the format of the data would make it easier to understand what is happening and why?

File: handlers/ticket-tailor-webhook/src/index.ts [hunk 4]
```diff
@@ -1,15 +1,39 @@
...
+import type { Stage } from '@modules/stage';
+import type { APIGatewayProxyResult, Handler, SQSEvent } from 'aws-lambda';
+import { getWebhookValidationSecret } from './hMacKey';
+import type { Payload } from './verifySignature';
+import { hasMatchingSignature } from './verifySignature';
 
 export const handler: Handler = async (
-	event: APIGatewayProxyEvent,
+	event: SQSEvent,
 ): Promise<APIGatewayProxyResult> => {
```
Comment (MaelGNM) [inline handlers/ticket-tailor-webhook/src/index.ts:9]: This lambda doesn't need to return and `APIGatewayProxyResult` so I think you could simplify things by changing the return type

File: handlers/ticket-tailor-webhook/src/verifySignature.ts (hunk 3 above)
Comment (Joemgru) [reply handlers/ticket-tailor-webhook/src/verifySignature.ts]: Good idea, I've put in a comment and also changed the logic a bit to hopefully make it clearer.

**Review by MaelGNM — APPROVED** [review]
Looks good to me

**Pull request merged**